libdnf5/base/goal.cpp

```diff
@@ -2725,9 +2722,9 @@ void Goal::Impl::add_group_upgrade_to_goal(
             old_set.emplace(pkg.get_name());
         }
         // set of package names that are part of the available version of the group
-        std::set<std::string> new_set{};
+        std::vector<std::string> new_set{};
         for (const auto & pkg : available_group.get_packages()) {
```

Comment:
It would probably be better to provide size hint to avoid repeated reallocations, but not sure if it's worth it:
```suggestion
        const auto & pkgs_from_available_group = available_group.get_packages();

        std::vector<std::string> new_set;
        new_set.reserve(pkgs_from_available_group.size());
        for (const auto & pkg : pkgs_from_available_group) {
```

Reply:
Makes sense, I changed it according to your suggestion. Thanks!